### DISCUSSION TITLE
quarkus.config.locations : Can I load all application.properties files under a given classpath directory recursively ?

### DISCUSSION THREAD

**Discussion by omasseau** [quarkusio/quarkus#41863] [Q&A] [answered]
Hello,

The documentation (https://quarkus.io/guides/config-reference#locations) says that we can provide a directory to quarkus.config.locations.
So I thought it would be possible to automatically scan all .properties files under a given directory  found in the classpath as additional locations for configuration properties files.
ex: 
`quarkus.config.locations=tenants/` (with tenants directory being in /resources/).

But it seems it is not possible. I have to expicitely define the path of each .properties file under the directory : 
`quarkus.config.locations=tenants/fr/authentication.properties,tenants/fr/datasources.properties,tenants/lu/authentication.properties,tenants/lu/datasources.properties`

It works but can be problematic when you may later have a lot of additional tenants.

What I would like to be is be able to define directories for each tenant of my application, each directory containing the properties associated to this tenant.
Example : 

![image](https://github.com/user-attachments/assets/0c59532a-efd5-432c-9f81-0938c34737e5)

I also tried to use wildcards in the path but it seems it is not supported also. Can you confirm ?
ex : 
`quarkus.config.locations=tenants/*/*.properties`

If it is not supported, I think it would be a great thing to support in the future :)

EDIT:
After some tests it seems directory only works if using a file: URI (and only application.properties files are loaded)
ex : 
`quarkus.config.locations=file:/C:/dev/git/test/src/main/resources/tenants/fr,file:/C:/dev/git/test/src/main/resources/tenants/lu`

But still not recursive, so I cannot just do : 
`quarkus.config.locations=file:/C:/dev/git/test/src/main/resources/tenants`

[a comment by an automated account is not shown]

**Reply by radcortez** [reply, marked as answer]
Yes, wildcards and directory recursion are not supported. You can open an enhancement request at https://github.com/smallrye/smallrye-config/.

>  After some tests it seems directory only works if using a file: URI (and only application.properties files are loaded)

Relative path works to the working directory

**Reply by omasseau** [reply]
Enhancement request created : https://github.com/smallrye/smallrye-config/issues/1197 ;)
